src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -1504,8 +1504,33 @@ public Response getSiteMigrationInBulkUpload(
 			for (String siteId : bulkUploadSiteIds) {
 				// for each site id, start the migration process
 				// associate it with the bulk id
+				
 				// 1. get site name
 				String siteName = getSiteName(request, siteId);
+				if (siteName == null)
+				{
+					// if the site id is invalid, and we cannot find the site
+					// generate an empty migration record with error and move on
+					
+					/* Timestamp start_time, Timestamp end_time,
+					String destination_type, String destination_url, String status*/
+					Migration migrationWithWrongSiteId = new Migration(bulkMigrationId, bulkMigrationName,
+							siteId, "default_site_name"/*site name*/,
+							"default_tool_id"/*tool id*/, "default_tool_name"/*tool name*/,
+							userId/*migrated by*/,
+							new java.sql.Timestamp(System.currentTimeMillis()), new java.sql.Timestamp(System.currentTimeMillis()),
+							Utils.MIGRATION_TYPE_BOX, "",
+							"Cannnot find a CTools site with the site id.");
+					try {
+						repository.save(migrationWithWrongSiteId);
+					} catch (IllegalArgumentException e) {
+						log.error("Exception " + migrationWithWrongSiteId.toString() + e.getMessage());
+					} catch (Exception e) {
+						log.error("Exception " + migrationWithWrongSiteId.toString() + e.getMessage());
+					}
+					break;
```

Comment:
I see that if we don't get valid `siteName` then we are exiting the bulk upload process using the `break`. but why do we want to stop can we just continue with the next siteId in the list?.